state-chain/pallets/cf-elections/src/electoral_systems/liveness.rs

```diff
@@ -32,6 +32,24 @@ pub trait OnCheckComplete<ValidatorId> {
 	fn on_check_complete(validator_ids: BTreeSet<ValidatorId>);
 }
 
+#[macro_export]
+macro_rules! create_on_check_complete_hook {
+	($name:ident, $chain:expr) => {
```

Comment:
Could just provide the chain name, and then derive `ForeignChain::<chain_name>` and the name of the hook?